cg/apps/demultiplex/sample_sheet/override_cycles_validator.py

```diff
@@ -0,0 +1,172 @@
+import logging
+import re
+
+from cg.constants.demultiplexing import (
+    FORWARD_INDEX_CYCLE_PATTERN,
+    REVERSE_INDEX_CYCLE_PATTERN,
+    SampleSheetBCLConvertSections,
+)
+from cg.exc import OverrideCyclesError
+
+LOG = logging.getLogger(__name__)
+
+
+class OverrideCyclesValidator:
+    """Class for validating the override cycles value of a sample in a sample sheet."""
+
+    def __init__(
+        self,
+        run_read1_cycles: int | None = None,
+        run_read2_cycles: int | None = None,
+        run_index1_cycles: int | None = None,
+        run_index2_cycles: int | None = None,
+        is_reverse_complement: bool | None = None,
```

Comment:
Design consideration: don't pass instance specific data to the constructor of a class that performs an action (validation in this case).

Could this be done with Pydantic models instead?

Reply:
yes, but let's leave the pydantic model for another PR

Reply:
#2963